fast_barcode_scanner_platform_interface/lib/src/types/preview_configuration.dart

```diff
@@ -102,3 +102,46 @@ class PreviewConfiguration {
     return 'PreviewConfiguration{width: $width, height: $height, targetRotation: $targetRotation, textureId: $textureId, analysisResolution: $analysisResolution, analysisWidth: $analysisWidth, analysisHeight: $analysisHeight}';
   }
 }
+
+abstract class IOSApiMode {
+  static const avFoundation = AVFoundationMode();
+  static const visionStandard = VisionMode();
+
+  const IOSApiMode();
+
+  String get name;
+
+  Map<String, dynamic> get configMap => {
+        "apiMode": name,
+        ...extraConfig,
+      };
+
+  Map<String, dynamic> get extraConfig => {};
+}
+
+class AVFoundationMode extends IOSApiMode {
+  @override
+  final String name = "avFoundation";
+
+  const AVFoundationMode();
+}
+
+class VisionMode extends IOSApiMode {
+  static const standardConfidence = 0.6;
+
+  /// The minimum confidence that the Vision API should use to filter scanned
+  /// codes given as a number between 0..1
+  ///
+  /// defaults to 0.6
+  final double confidence;
+
+  @override
+  final String name = "vision";
+
+  const VisionMode({this.confidence = standardConfidence});
+
+  @override
+  Map<String, dynamic> get extraConfig => {
```

Comment:
I don't think the term "extra" does not provide any insight into what this config is or does. Can be removed.